v2/thirdpartyemailpassword/migration/account-creation/ep-migration-without-password-hash.mdx

```diff
@@ -0,0 +1,284 @@
+
+                                // If user is succesfully created  we will map the legacy id to their SuperTokens Id. 
+                                await SuperTokens.createUserIdMapping({
+                                    superTokensUserId: signupResponse.user.id,
+                                    externalUserId: legacyUserData.user_id
+                                })
+
+                                // We will also need to set the email verification status of the user
+                                if (legacyUserData.isEmailVerified) {
+                                    // Generate an email verification token for the user
+                                    let generateEmailVerificationTokenResponse = await EmailVerification.createEmailVerificationToken(legacyUserData.user_id, email, input.userContext);
+
+                                    if (generateEmailVerificationTokenResponse.status === "OK") {
+                                        // Verify the user's email
+                                        await EmailVerification.verifyEmailUsingToken(generateEmailVerificationTokenResponse.token, input.userContext);
+                                    }
+                                }
+                            } else {
+                                throw new Error("Should never come here")
+                            }
+                        }
+                    }
+                    return await originalImplementation.generatePasswordResetTokenPOST(input);
+                },
+            }
+        }
+    }
+})
+
+async function generatePassword(): Promise<string> {
+    // TODO: generate a random password 
+    return ""
+}
+
+async function retrieveUserDataFromExternalProvider(email: string): Promise<{
+    user_id: string,
+    isEmailVerified: boolean
+} | undefined> {
+    // TODO: retrieve user data if a user with the input email exists in the external provider.
+    return undefined;
+}
+```
+</TabItem>
+</BackendSDKTabs>
+
+The code above overrides the `generatePasswordResetTokenPOST` API with the following changes to achieve "**just in time**" migration:
+- Similar to [Step 1](#step-1-override-the-login-flow), we need to determine whether to migrate the user or not. To do this we check if the user exists in the external provider and SuperTokens.
+If the user exists in the external provider and does not exist in SuperTokens, we can determine that this user needs to be migrated.
+- The next step is to create a SuperTokens account with a temporary password, the password can be a random string since it will be reset by the user when they complete the reset password flow.
+- We now map the external `userId` to the SuperTokens `userId`. This will allow SuperTokens functions to refrence the user with the external `userId`.
+- Finally, depending on the email verification status of the user in the external provider we will also verify the user's email in SuperTokens.
+
```

Comment:
Add a section:
```
## When can I stop using my legacy auth provider?
```

Reply:
- wait for sometime - maybe after a certain percentage of users have migrated or certain time has passed
- remove overrides
- take left over users' emails and call the EmailPassword.signUp function with that and a random, strong password
- Encourage those users to go through the reset password flow.